### PR TITLE
Add Qwen3.6-35B-A3B (Qwen3.5-MoE-VL) support with Olive INT4 example

### PR DESCRIPTION
**Pull request by justinchuby**
Qwen3.6-35B-A3B is the vision-language variant of Qwen3.5-MoE. The HF
config reports ``model_type=qwen3_5_moe`` with both ``text_config`` and
``vision_config``; mobius dispatches it to a new ``qwen3_5_moe_vl``
registry entry that composes the existing ``Qwen35MoETextModel``
backbone with the Qwen3-VL ViT + ``InputMixer`` (same pipeline as the
dense Qwen3.5-VL ``Qwen35VL3ModelCausalLMModel``).

## Changes

- **Model:** ``Qwen35MoEVLDecoderModel`` + ``Qwen35MoEVL3ModelCausalLMModel``
  in ``src/mobius/models/qwen35.py``. Extended ``Qwen35MoETextModel.forward``
  to accept ``inputs_embeds`` for VL decoding. ``preprocess_weights``
  unpacks HF's fused ``experts.{gate_up,down}_proj`` tensors into
  per-expert names (HF stores ``[E, 2*inter, H]`` / ``[E, H, inter]``).
- **Dispatch:** ``_builder.py`` overrides ``model_type`` to
  ``qwen3_5_moe_vl`` when the composite config has a ``vision_config``;
  text-only Qwen3.5-MoE still resolves to ``qwen3_5_moe``.
- **Registry:** new ``qwen3_5_moe_vl`` entry with task ``hybrid-qwen-vl``.
- **L1:** tiny config + checker skip entry — ``build_graph_test.py`` passes.
- **L4/L5:** golden case + HF reference outputs for
  ``Qwen/Qwen3.6-35B-A3B`` (top-1 token=760 ""The"", 24-token generation
  captured).
- **Example:** ``examples/qwen3_6_35b_a3b_ort_genai.py`` runs ORT GenAI
  inference and supports ``--quantize {Q4_K_M,NF4}`` via Olive.

## Test status

- ✅ L1 ``build_graph_test`` for ``qwen3_5_moe_vl``
- ✅ L2 ``arch_validation``
- ✅ Real-weight build (131GB f32 decoder at ``/tmp/qwen3_6-onnx``)
- ✅ HF golden generation
- ⏳ L4/L5 against built ONNX (in progress)
- ⏳ Olive Q4_K_M quantization (running)

## Follow-ups

- bf16 + cuda EP builds + L4/L5 verification on quantized output
- Upload to HuggingFace

### CODE REVIEW

**Comment by justinchuby** [comment]
## Verification update

✅ **Olive Q4_K_M quantization** — decoder shrank from 128 GB (f32) to **21 GB**; quantization ran in 17 minutes (CPU, k-quant block_size=32).

✅ **L5 smoke test** — assembled `/tmp/qwen3_6-onnx-Q4_K_M/` with quantized decoder + f32 embedding/vision_encoder + tokenizer, ran the example script with the cat-chonk image:

```
Prompt: Describe this image in one short sentence.
Output: The user wants a short sentence describing the image.\n\n1
```

Identical token-for-token to the first 12 tokens of the HF bf16 golden reference (`testdata/golden/vision-language/qwen3_6-35b-a3b_generation.json`).

🔧 **Fix included**: Olive's `OnnxKQuantQuantization` renames `logits` → `logits_Q4` in the decoder output. `_run_session` now aliases it back so the same generation loop works for both quantized and unquantized models (commit bfb3bfc).

### Still TODO (out of this PR's scope)
- CUDA EP build (`--ep cuda`)
- bf16 build variant
- Upload to HuggingFace Hub

File: tests/e2e_golden_test.py [hunk 1]
```diff
@@ -868,7 +874,12 @@ def _run_vl_generation(
 
         # --- Step 5: decode loop ---
         # Embed each new token through the embedding model with no image features.
-        empty_image = np.zeros((0, hidden_size), dtype=np.float32)
+        image_feat_dtype = (
+            emb_session.get_input_dtype(image_feat_input)
+            if image_feat_input is not None
+            else None
+        ) or np.float32
+        empty_image = np.zeros((0, hidden_size), dtype=image_feat_dtype)
```
Comment (justinchuby) [inline tests/e2e_golden_test.py:882]: Why isn't this always f32?

File: src/mobius/models/qwen35.py [hunk 2]
```diff
@@ -569,3 +576,159 @@ def preprocess_weights(
...
+            if stripped.startswith("model."):
+                stripped = stripped[len("model.") :]
+            if stripped.startswith("visual."):
+                continue
+            if stripped.startswith("language_model."):
+                stripped = stripped[len("language_model.") :]
+            renamed[stripped] = value
+        if self.config.tie_word_embeddings:
+            if "lm_head.weight" not in renamed and "model.embed_tokens.weight" in renamed:
+                renamed["lm_head.weight"] = renamed["model.embed_tokens.weight"]
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply src/mobius/models/qwen35.py]: Good catch — confirmed unreachable. Qwen35MoEVLDecoderModel is constructed only as a sub-module of Qwen35MoEVL3ModelCausalLMModel (line ~667) and never registered standalone, so this preprocess_weights was dead code, hiding both bugs you flagged (the un-stripped key check, and the missing fused MoE expert unpack). Dropped the whole method in 5baaa24 with a comment explaining the routing lives on the wrapper.

File: src/mobius/models/qwen35.py [hunk 3]
```diff
@@ -569,3 +576,159 @@ def preprocess_weights(
...
+                continue
+            stripped = key
+            if stripped.startswith("model."):
+                stripped = stripped[len("model.") :]
+            if stripped.startswith("visual."):
+                continue
+            if stripped.startswith("language_model."):
+                stripped = stripped[len("language_model.") :]
+            renamed[stripped] = value
+        if self.config.tie_word_embeddings:
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply src/mobius/models/qwen35.py]: Good catch — confirmed unreachable. Qwen35MoEVLDecoderModel is constructed only as a sub-module of Qwen35MoEVL3ModelCausalLMModel (line ~667) and never registered standalone, so this preprocess_weights was dead code, hiding both bugs you flagged (the un-stripped key check, and the missing fused MoE expert unpack). Dropped the whole method in 5baaa24 with a comment explaining the routing lives on the wrapper.

File: examples/qwen3_6_35b_a3b_ort_genai.py [hunk 4]
```diff
@@ -0,0 +1,1124 @@
...
+DeltaNet recurrent state, so generation here uses manual ONNX Runtime
+sessions with onnxruntime-genai providing only the tokenizer.
+
+Requirements::
+
+    pip install mobius-ai[ort-genai]
+    # For INT4 quantization (Olive from source recommended):
+    pip install -e /path/to/Olive cupy-cuda12x
+
+Usage::
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply examples/qwen3_6_35b_a3b_ort_genai.py:29]: Right, ort-genai extras didn't include Pillow but the example scripts import PIL.Image for --image. Fixed in 5baaa24 by adding Pillow to the ort-genai extras (matches what testing extras already do).
